File: ios/HealthCareApp/EventEmitter/DeviceEventEmitter.m
//
//  DeviceEventEmitter.m
//  HealthCareApp
//
//  Created by Vivek Singh on 24/06/20.
//  Copyright © 2020 Facebook. All rights reserved.
//
#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"
@interface RCT_EXTERN_MODULE(DeviceEventEmitter, RCTEventEmitter)
RCT_EXTERN_METHOD(supportedEvents)

@end

File: ios/HealthCareApp/EventEmitter/DeviceEventEmitter.swift
//
//  DeviceEventEmitter.swift
//  HealthCareApp
//
//  Created by Vivek Singh on 24/06/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation
@objc(DeviceEventEmitter) class DeviceEventEmitter:RCTEventEmitter{
        
    override init() {
        super.init()
        EventEmitter.shared.emitter = self
    }
    
    override func supportedEvents() -> [String]! {
        EventEmitter.shared.emitter = self
       return EventEmitter.shared.allEvents
    }
    
    override static func requiresMainQueueSetup() -> Bool {
       return false
    }
    
}
